addons/hr_expense/hr_expense.py

```diff
@@ -65,257 +32,219 @@ def _get_currency(self, cr, uid, context=None):
+    note = fields.Text()
+    amount_untaxed = fields.Float(compute='_amount', string='Untaxed Amount', digits=dp.get_precision('Account'), store=True)
+    amount_tax = fields.Float(compute='_amount', string='Taxes', digits=dp.get_precision('Account'), store=True)
+    amount = fields.Float(compute='_amount', string='Total', digits=dp.get_precision('Account'), store=True)
+    currency_id = fields.Many2one('res.currency', string='Currency', required=True, readonly=True, states={'draft': [('readonly', False)], 'confirm': [('readonly', False)]}, default=lambda self:self.env.user.company_id.currency_id.id)
+    department_id = fields.Many2one('hr.department', string='Department', readonly=True, states={'draft': [('readonly', False)], 'confirm': [('readonly', False)]})
+    company_id = fields.Many2one('res.company', string='Company', required=True, readonly=True, states={'draft': [('readonly', False)], 'confirm':[('readonly', False)]}, default=lambda self: self.env.user.company_id)
+    state = fields.Selection([
+        ('draft', 'New'),
+        ('cancelled', 'Refused'),
+        ('confirm', 'Waiting Approval'),
+        ('accepted', 'Approved'),
+        ('done', 'Waiting Payment'),
+        ('paid', 'Paid'),
+        ],
+        string='Status', readonly=True, track_visibility='onchange', copy=False, default='draft', required=True,
+        help='When the expense request is created the status is \'Draft\'.\n It is confirmed by the user and request is sent to admin, the status is \'Waiting Confirmation\'.\
+        \nIf the admin accepts it, the status is \'Accepted\'.\n If the accounting entries are made for the expense request, the status is \'Waiting Payment\'.')
+
+    @api.multi
+    def unlink(self):
+        for sheet in self:
+            if sheet.state != 'draft':
+                raise Warning(_('You can only delete draft expenses!'))
+        return super(hr_expense_sheet, self).unlink()
+
+    @api.onchange('currency_id')
+    def onchange_currency_id(self):
+        account_journal = self.env['account.journal'].search([('type', '=', 'purchase'), ('currency', '=', self.currency_id.id), ('company_id', '=', self.company_id.id)], limit=1)
+        self.journal_id = account_journal.id
+
+    @api.multi
+    def sheet_draft(self):
+        for sheet in self:
+            sheet.line_ids.expense_line_draft()
+        return self.write({'state': 'draft'})
+
+    @api.multi
+    def sheet_confirm(self):
+        for sheet in self:
+            if not sheet.line_ids:
+                raise Warning(_('You cannot submit expense which has no expense line.'))
+            if sheet.employee_id and sheet.employee_id.parent_id.user_id:
+                self.message_subscribe_users(user_ids = sheet.employee_id.parent_id.user_id.ids)
+            if False not in [True if expense.state in ['draft'] else False for expense in sheet.line_ids]:
+                sheet.line_ids.write({'state': 'confirm'})
```

Comment:
@rim-odoo @mdi-odoo 
If expense sheet is in 'draft' state by "set to draft" button and 'hr manager' is want 'Waiting Approval' state on expense sheet so inside sheet expense's is also 'submitted' state.